networking/external_dns_operator/nw-control-dns-records-public-hosted-zone-aws.adoc

```diff
@@ -0,0 +1,49 @@
+:_content-type: PROCEDURE
+[id="nw-control-dns-records-public-hosted-zone-aws_{context}"]
+= Creating DNS records on an public hosted zone for AWS by using Red Hat External DNS Operator
+
+You can create DNS records on public hosted zone for AWS by using Red Hat External DNS Operator.
+
+.Procedure
+
+. Export the `kubeconfig` file:
++
+[source,terminal]
+----
+$ export KUBECONFIG=~/<path to>/kubeconfig
+----
+
+. To create `ExternalDNS` CR for `route`, run the following command:
++
+[source,yaml]
+----
+apiVersion: externaldns.olm.openshift.io/v1alpha1
+kind: ExternalDNS
+metadata:
+  name: sample-aws <1>
+spec:
+  domains:
+  - filterType: Include   <2>
+    matchType: Exact   <3>
+    name: testextdnsoperator.apacshift.support <4>
+    provider:
+      type: AWS <5>
+    source:  <6>
+      type: OpenShiftRoute <7>
+      openshiftRouteOptions:
+        routerName: default <8>
+----
+<1> The `name` field defines the name of external DNS CR.
+<2> By default all hosted zones are selected as potential targets. You can include a hosted zone that you need.
+<3> The matching of the target zone's domain has to be exact (as opposed to regular expression match).
+<4> Specify the exact domain of the zone you want to update. The hostname of the routes must be subdomains of the specified domain.
+<5> The `type` field defines the `AWS Route53` DNS provider.
+<6> The `source` field defines an option for route source.
+<7> The `type` field defines source route. External DNS Operator creates DNS records for routes on your cloud DNS.
+<8> If the source is `openshift-route` then you can pass the ingress controller name. External DNS Operator selects the canonical hostname of a router as the target while creating CNAME record.
+
+. Check the *AWS Console > Route53* page to ensure you added the records to the hosted zone.
+
+. Run `nslookup/dig` command to check if you can resolve `service-name.<project>.svc.cluster.local` to the service ClusterIP.
```

Comment:
From above CR, looks it just can resolve route to canonical hostname of the router but not the one to resolve `service-name.<project>.svc.cluster.local` to the service ClusterIP.

@miheer Could you help confirm? Thanks

Reply:
Addressed